modules/aws/vault/vault-iam.tf

```diff
@@ -38,14 +41,16 @@ data "aws_iam_policy_document" "vault-kms-unseal" {
 }
 
 resource "aws_iam_role_policy" "vault-kms-unseal" {
+  count  = "${var.vault_auto_unseal ? 1 : 0}"
   name   = "${var.cluster_name}-vault-kms-unseal"
   role   = "${aws_iam_role.vault.id}"
   policy = "${data.aws_iam_policy_document.vault-kms-unseal.json}"
 }
 
 resource "aws_iam_role_policy" "vault-s3-ignition" {
-  name = "${var.cluster_name}-vault-s3-ignition"
-  role = "${aws_iam_role.vault.id}"
+  count = "${var.vault_auto_unseal ? 1 : 0}"
```

Comment:
same for s3. I think it is completely fine to provision vault in the same way as in other regions